### PR TITLE
Closes #12 

### LINKED ISSUE #12: Build and packaging from github actions
**Describe the bug**
At present, there is no way to build and push the packages to github registry. This prevents the further steps in deployment.

**Expected behavior**
On merge of a PR

- The version in POM has to be rev'd
- The jar has to be pushed to the GitHub registry
 - Any dependencies would have to be rev'd and built

### PR DESCRIPTION
**Pull request by aprajshekhar**
Fixes #12 

## Proposed Changes

  - Update pom version
  - Push to repsy repo
  - Create git tag

### CODE REVIEW

File: .github/workflows/maven-publish.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        distribution: 'adopt'
+#     - name: Download Maven
+#       run: |
+#         curl -sL https://www-eu.apache.org/dist/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.zip -o maven.zip
+#         apt-get update
+#         apt-get -y install unzip
+#         unzip -d /usr/share maven.zip
+#         rm maven.zip
+#         ln -s /usr/share/apache-maven-3.6.3/bin/mvn /usr/bin/mvn
+#         echo "M2_HOME=/usr/share/apache-maven-3.6.3" | tee -a /etc/environment
```
Comment (hybridx) [inline .github/workflows/maven-publish.yml:37]: @aprajshekhar Is this going to be required in the future? If not I suppose we can remove it

**Review by hybridx — CHANGES_REQUESTED** [review]
@aprajshekhar Not sure if this is a correct question, but I would like to know if class files can be built on github actions server and use to deploy there itself. If that's the case then I suppose we can remove them from this PR which will reduce the number of files and . We can also add `*.class` in gitignore. Correct me if I'm wrong

File: .github/workflows/maven-publish.yml (hunk 1 above)
Comment (aprajshekhar) [reply .github/workflows/maven-publish.yml:37]: @hybridx we might need it if the version supported by action changes. I would like it to be here as a reference, if required in the future.

**Comment by aprajshekhar** [comment]
> Not sure if this is a correct question, but I would like to know if class files can be built on github actions server and use to deploy there itself. If that's the case then I suppose we can remove them from this PR which will reduce the number of files and . We can also add *.class in gitignore. Correct me if I'm wrong

The class file should never be a part of codebase. The .bin file somehow got added. .class is already in .gitignore. I have already updated .gitignore for bin folder. However, the pushed bin folder was not deleted. The deletion got pushed as a part of this PR. Sorry for that.

**Review by hybridx — APPROVED** [review]
LGTM

**Review by hybridx — APPROVED** [review]

**Pull request merged**